projects/fundamental-ngx/src/lib/date-picker/date-picker.component.ts

```diff
@@ -0,0 +1,62 @@
+import { Component, Input, OnInit, HostListener } from '@angular/core';
+import { DatePickerType } from '../calendar/calendar.component';
+
+@Component({
+    selector: 'fd-date-picker',
+    templateUrl: './date-picker.component.html',
+    styleUrls: ['./date-picker.component.scss']
+})
+export class DatePickerComponent implements OnInit {
+    @Input() type: DatePickerType = 'single';
+    inputFieldDate = null;
+    isValidDateInput: boolean = false;
+    isOpen: boolean = false;
+    dateFromDatePicker: string = '';
+
+    openCalendar(e) {
+        this.isOpen = !this.isOpen;
+        this.getInputValue(e);
+        if (this.isValidDateInput) {
+            this.inputFieldDate = null;
+        }
+    }
+
+    closeCalendar() {
+        if (this.isOpen) {
```

Comment:
Clicking outside of the component does not close the calendar, take a look at popover.component.ts to see how this is handled with the time picker

Reply:
Yes, I used this implementation, but then, when you click on the day (select day), it closes the calendar. Should think of other way to handle it.

Reply:
Fixed. Now if you click outside the calendar, it closes